pandas/util/testing.py

```diff
@@ -1047,6 +1051,14 @@ def assert_interval_array_equal(left, right, exact='equiv',
     assert_attr_equal('closed', left, right, obj=obj)
 
 
+def assert_period_array_equal(left, right, obj='PeriodArray'):
```

Comment:
Should we rather add a general `assert_array_equal` which can deal with all our own Array classes? 
(but again, not necessarily for this PR)

Reply:
Are people opposed to singledispatch?

```python
In [1]: import pandas as pd

In [2]: import pandas.util.testing as tm

In [3]: from functools import singledispatch

In [4]: @singledispatch
   ...: def assert_equal(left, right, **kwargs):
   ...:     pass  # probably raise for unidentified type

In [5]: @assert_equal.register(pd.Series)
   ...: def _(left, right, **kwargs):
   ...:     return tm.assert_series_equal(left, right, **kwargs)

In [6]: @assert_equal.register(pd.DataFrame)
   ...: def _(left, right, **kwargs):
   ...:     return tm.assert_frame_equal(left, right, **kwargs)
```

this would let 3rd part EAs register their own `assert_my_EA_equal` checks, and we can use them without having to know about them ahead of time.

Reply:
(again, not for this PR, and I think singledispatch is new in Py3, so not till the new year).